ice-adapter/src/main/java/com/faforever/iceadapter/ice/GameSession.java

```diff
@@ -52,6 +52,9 @@ public GameSession() {}
      * @return the port the ice adapter will be listening/sending for FA
      */
     public int connectToPeer(String remotePlayerLogin, int remotePlayerId, boolean offer, int preferredPort) {
+        if (peers.containsKey(remotePlayerId)) {
+            disconnectFromPeer(remotePlayerId);
```

Comment:
Hmm. Should this not trigger a reconnectToPeer instead (along with a warning log)?
And the remaining code goes into an else block to avoid infinite recursion.